pkg/ublox/ublox.go

```diff
@@ -0,0 +1,165 @@
+	}
+	if e := u.Init(); e != nil {
+		glog.Errorf("error creating UBlox object %s", e)
+		return nil, e
+	}
+	return u, nil
+}
+func NewMockUblox(fn func(cmdStr string) ([]byte, error)) (*UBlox, error) {
+	u := &UBlox{
+		protoVersion: nil,
+		mockExp:      fn,
+	}
+	return u, nil
+}
+
+/*
+ ENVIRONMENT:
+#    Options in the UBXOPTS environment variable will be parsed before
+#    the CLI options.  A handy place to put your '-f /dev/ttyXX -s SPEED'
+#
+# To see what constellations are enabled:
+#       ubxtool -p CFG-GNSS -f /dev/ttyXX
+#
+# To disable GLONASS and enable GALILEO:
+#       ubxtool -d GLONASS -f /dev/ttyXX
+#       ubxtool -e GALILEO -f /dev/ttyXX
+#
+
+*/
+
+// Init ...
+func (u *UBlox) Init() (err error) {
+	var stdout []byte
+	var protoVersion float64
+
+	cmdLine := fmt.Sprintf("ubxtool %s", CMD_PROTO_VERSION)
+	args := strings.Split(cmdLine, " ")
+
+	glog.Infof("/usr/bin/bash", "-c", fmt.Sprintf("\"ubxtool %s\"", CMD_PROTO_VERSION))
+	if stdout, err = exec.Command(args[0], args[1:]...).Output(); err == nil {
+		glog.Infof("ubxtool command output ..%s", string(stdout))
+		if protoVersion, err = FindProtoVersion(string(stdout)); err == nil {
+			u.protoVersion = &protoVersion
+		} else {
+			err = fmt.Errorf("UBlox could not find version for method %s with error %s", "MON_VER", err)
+		}
+	} else {
+		err = fmt.Errorf("UBlox could not find version for method %s with error %s", "MON-VER", err)
+	}
+	return
+}
+
+// FindProtoVersion ...
+func FindProtoVersion(stdout string) (float64, error) {
+	var parseError error
+	var protoVersion float64
+	match := PorotoVersionRegEx.FindStringSubmatch(string(stdout))
+	if match != nil {
+		if len(match) > 1 {
+			protoVersion, parseError = strconv.ParseFloat(match[1], 2)
+
+		}
+	}
+	return protoVersion, parseError
+}
+func (u *UBlox) query(command string) ([]byte, error) {
+	if u.protoVersion == nil {
+		return []byte{}, fmt.Errorf("Cannot query UBlox without protocol version ")
+	}
+	stdout, err := exec.Command("/usr/bin/bash", "-c", fmt.Sprintf("\"ubxtool  -v 1 -P %f  -p %s\"", *u.protoVersion, command)).Output()
+	return stdout, err
+}
+
+// EnableDisableVoltageController ... enable disable voltage controler 1 or 0
+// Enable GNSS Antenna voltage control
+// # ubxtool -v 1  -P 29.20 -z CFG-HW-ANT_CFG_VOLTCTRL,1
+// connected to tcp://localhost:2947
+// sent:
+// UBX-CFG-VALSET:
+// version 0 layer 0x7 transaction 0x0 reserved 0
+// layers (ram bbr flash) transaction (Transactionless)
+// item CFG-HW-ANT_CFG_VOLTCTRL/0x10a3002e val 1
+
+// UBX-ACK-ACK:
+// ACK to Class x06 (CFG) ID x8a (VALSET)
+// TODO: Should read ACK-ACK to confirm right and read the item
+func (u *UBlox) EnableDisableVoltageController(command string, value int) ([]byte, error) {
+	if u.protoVersion == nil {
+		return []byte{}, fmt.Errorf("Cannot query UBlox without protocol version ")
+	}
+	commandArgs := []string{"/usr/bin/bash", "-c", fmt.Sprintf("\"ubxtool  -v 1 -P %f  -p %s,%d\"", *u.protoVersion, command, value)}
```

Comment:
should `-p` be part of command?  A command could be `-p ...` or `-z ...`.

Reply:
This is invalid , I have updated the code